internal/compiler-bridge/src/main/scala/xsbt/CompilerInterface.scala

```diff
@@ -187,14 +187,6 @@ private final class CachedCompiler0(args: Array[String], output: Output, initial
   }
 
   def processUnreportedWarnings(run: compiler.Run): Unit = {
-    // allConditionalWarnings and the ConditionalWarning class are only in 2.10+
-    final class CondWarnCompat(
-        val what: String,
-        val warnings: mutable.ListBuffer[(compiler.Position, String)]
-    )
-    implicit def compat(run: AnyRef): Compat = new Compat
-    final class Compat { def allConditionalWarnings = List[CondWarnCompat]() }
-
     val warnings = run.allConditionalWarnings
```

Comment:
It looks like this code has been doing the wrong thing since Scala 2.11, after this change we get:
```scala
> compilerBridgeJVM2_10/compile
...
[info] Done compiling.
> compilerBridgeJVM2_11/compile
[error] /home/smarter/opt/zinc/internal/compiler-bridge/src/main/scala/xsbt/CompilerInterface.scala:190:24: value allConditionalWarnings is not a member of CachedCompiler0.this.compiler.Run
[error]     val warnings = run.allConditionalWarnings
[error]                        ^
```

Reply:
In 2.11, allCondiitonalWarnings was moved from Global to Reporting: https://github.com/scala/scala/commit/98216be3f3e546fc320ab5182ac5c129707db1ce. So we would need another piece of compat code to handle this properly. I don't actually know what this `processUnreportedWarnings` thing is trying to do, why are we trying to report unreported things? Can someone come up with a testcase that exercises it?

Reply:
I tried to dig into this, but I really have no idea what any of this code is trying to do so this is the point where I give up.

Reply:
Here's the commit which introduced processUnreportedWarnings for some background for anyone who wants to pick this up: https://github.com/sbt/zinc/commit/7440fbcec147211694ba0c79d432dc8923dbcec6 and the commit which added the compat stuff: https://github.com/sbt/zinc/commit/44330cf1b13a142457a6f1ee23d6dca1e35b51fe